core/domain/state_domain_test.py

```diff
@@ -1134,54 +1134,66 @@ def test_rte_content_validation_for_android(self):
-        init_state.update_interaction_id('DragAndDropSortInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('EndExploration')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('FractionInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('ItemSelectionInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('MathEquationInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('MultipleChoiceInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('NumberWithUnits')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('NumericInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('TextInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('NumericExpressionInput')
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-
-        # Empty interaction.
-        init_state.update_interaction_id(None)
-        self.assertTrue(init_state.interaction.is_supported_on_android_app())
-
-        # Invalid interactions.
-        init_state.update_interaction_id('CodeRepl')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('GraphInput')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('InteractiveMap')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('LogicProof')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('MusicNotesInput')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('PencilCodeEditor')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
-        init_state.update_interaction_id('SetInput')
-        self.assertFalse(init_state.interaction.is_supported_on_android_app())
+        _checked_interaction_ids = set()
+
+        def _create_init_state_for_interaction_verification():
+            """Creates an init state for interaction verification."""
+            exploration = (
+                exp_domain.Exploration.create_default_exploration('0'))
+            return exploration.states[exploration.init_state_name]
+
+        def _verify_interaction_supports_android(self, interaction_id):
```

Comment:
Here & below: do we still need ``self`` if these functions aren't part of a class? Prefer to remove them.

Reply:
I had to add it because we do a `self.assertTrue(..)` inside.